src/imgui/ImGuiMachine.cc

```diff
@@ -142,6 +172,33 @@ void ImGuiMachine::paintSelectMachine(const MSXMotherBoard* motherBoard)
 			ImGui::Separator();
 		}
 
+		if (!recentMachines.empty()) {
+			ImGui::TextUnformatted("Recently used machines:"sv);
+			auto* info = findMachineInfo(recentMachines.front());
+			im::Combo("##recent", info->displayName.c_str(), [&]{
+				for (const auto& item : recentMachines) {
+					info = findMachineInfo(item);
+					bool ok = info && getTestResult(*info).empty();
+					im::StyleColor(!ok, ImGuiCol_Text, getColor(imColor::ERROR), [&]{
+						bool selected = info->configName == newMachineConfig;
+						if (ImGui::Selectable(info->displayName.c_str(), selected)) {
+							newMachineConfig = info->configName;
+							if (ok) {
+								showSelectMachine = false; // close window
+								manager.executeDelayed(makeTclList("machine", newMachineConfig));
+								addRecentItem(recentMachines, newMachineConfig);
+							}
+						}
+						if (ImGui::IsItemHovered(ImGuiHoveredFlags_DelayNormal | ImGuiHoveredFlags_NoSharedDelay | ImGuiHoveredFlags_Stationary)) {
+							im::ItemTooltip([&]{
+								printConfigInfo(*info);
+							});
+						}
+					});
```

Comment:
Assuming you go for the alternative that `info` is guaranteed `!= nullptr`, then this block of code (starting after line 180, where you obtain `info`) is very similar to the block below in the `Available machines` list box. The main difference is double-click vs single-click, but that could become a parameter on the factored out code.

I propose to do this in a separate commit.

Reply:
OK, let's hope that I can more easily solve conflicts in case they appear when processing further feedback.

Should be fixed in latest version.